sdks/java/io/iceberg/build.gradle

```diff
@@ -54,11 +56,12 @@ dependencies {
     implementation "org.apache.iceberg:iceberg-parquet:$iceberg_version"
     implementation "org.apache.iceberg:iceberg-orc:$iceberg_version"
     implementation library.java.hadoop_common
+    // Hadoop GCS filesystem dependencies
+    runtimeOnly library.java.hadoop_client
```

Comment:
I _think_ I'm OK with this... it isn't very Java-like to put optional dependencies as mandatory dependencies though. Typically these should also be resolvable via transitive dependencies if we depend on e.g. GCS core which provides the GCS filesystem.